tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -237,7 +237,13 @@ function kbs_k8s_deploy() {
 
 	# Tests should fill kbs resources later, however, the deployment
 	# expects at least one secret served at install time.
-	echo "somesecret" > overlays/key.bin
+	if [ -d "overlays/$(uname -m)" ]; then
```

Comment:
Hi @BbolroC !

Kata uses trustee v0.9.0 which doesn't contain the multi-arch support. So once we bump to newest trustee we will re-do this change? i.e. to always `echo "somesecret" > overlays/$(uname -m)/key.bin` ? 

If yes, could we mark with a `TODO` comment?

Thinking aloud... perhaps trustee's `deploy-kbs.sh` could deal with that secret. Consumers should simply export the secret in a variable, this way any break on trustee side doesn't affect us.

Reply:
ah, also any special reason to not bump trustee on this PR?

Reply:
I guess that the trustee team cut the `v0.9.0` as they felt it was stable. That doesn't mean we can't bump it here, but we probably want to get some input/agreement with them first.

Reply:
Thanks for the comment. I think the change covers both cases (a version with/without the multi-arch support), which means x86_64 is not affected as the version changes. However, we could refactor the code if `echo "somesecret" > overlays/key.bin` is no longer used. 

> Thinking aloud... perhaps trustee's deploy-kbs.sh could deal with that secret. Consumers should simply export the secret in a variable, this way any break on trustee side doesn't affect us.

I was thinking the same. But even in this case, we will face the similar issue once when bumping the version between two repos.

ATM, the s390x nightly test is using the main branch for trustee like:

```diff
diff --git a/tests/integration/kubernetes/confidential_kbs.sh b/tests/integration/kubernetes/confidential_kbs.sh
index 36f4244b5..1e297eaf1 100644
--- a/tests/integration/kubernetes/confidential_kbs.sh
+++ b/tests/integration/kubernetes/confidential_kbs.sh
@@ -230,10 +230,10 @@ function kbs_k8s_deploy() {

        echo "::group::Clone the kbs sources"
        git clone --depth 1 "${repo}" "$COCO_TRUSTEE_DIR"
-       pushd "$COCO_TRUSTEE_DIR"
-       git fetch --depth=1 origin "${version}"
-       git checkout FETCH_HEAD -b kbs_$$
-       popd
+       #pushd "$COCO_TRUSTEE_DIR"
+       #git fetch --depth=1 origin "${version}"
+       #git checkout FETCH_HEAD -b kbs_$$
+       #popd
        echo "::endgroup::"

        pushd "${COCO_KBS_DIR}/config/kubernetes/"
```

I've opened a thread for the update of the trustee for kata containers on Slack: https://cloud-native.slack.com/archives/C039JSH0807/p1720080885154259. If the trustee team is okay to bump it, I will include it here. Thanks.